tiled/server/settings.py

```diff
@@ -1,82 +1,55 @@
-import collections
 import os
 import secrets
 from datetime import timedelta
 from functools import cache
 from typing import Any, List, Optional
 
-from pydantic_settings import BaseSettings
+from pydantic import Field
+from pydantic.dataclasses import dataclass
+from pydantic_settings import BaseSettings, SettingsConfigDict
 
-DatabaseSettings = collections.namedtuple(
-    "DatabaseSettings", "uri pool_size pool_pre_ping max_overflow"
-)
+
+# hashable cache key for use in tiled.authn_database.connection_pool
+@dataclass(unsafe_hash=True)
+class DatabaseSettings:
+    uri: Optional[str] = None
+    pool_size: int = 5
+    pool_pre_ping: bool = True
+    max_overflow: int = 5
 
 
 class Settings(BaseSettings):
     tree: Any = None
-    allow_anonymous_access: bool = bool(
-        int(os.getenv("TILED_ALLOW_ANONYMOUS_ACCESS", False))
-    )
-    allow_origins: List[str] = [
-        item for item in os.getenv("TILED_ALLOW_ORIGINS", "").split() if item
-    ]
+    allow_anonymous_access: bool = False
+    allow_origins: List[str] = Field(default_factory=list)
     authenticator: Any = None
     # These 'single user' settings are only applicable if authenticator is None.
-    single_user_api_key: str = os.getenv(
-        "TILED_SINGLE_USER_API_KEY", secrets.token_hex(32)
-    )
-    single_user_api_key_generated: bool = not (
-        "TILED_SINGLE_USER_API_KEY" in os.environ
-    )
-    # The TILED_SERVER_SECRET_KEYS may be a single key or a ;-separated list of
-    # keys to support key rotation. The first key will be used for encryption. Each
-    # key will be tried in turn for decryption.
-    secret_keys: List[str] = os.getenv(
-        "TILED_SERVER_SECRET_KEYS", secrets.token_hex(32)
-    ).split(";")
-    access_token_max_age: timedelta = timedelta(
-        seconds=int(os.getenv("TILED_ACCESS_TOKEN_MAX_AGE", 15 * 60))  # 15 minutes
-    )
-    refresh_token_max_age: timedelta = timedelta(
-        seconds=int(
-            os.getenv("TILED_REFRESH_TOKEN_MAX_AGE", 7 * 24 * 60 * 60)
-        )  # 7 days
-    )
-    session_max_age: Optional[timedelta] = timedelta(
-        seconds=int(os.getenv("TILED_SESSION_MAX_AGE", 365 * 24 * 60 * 60))  # 365 days
-    )
+    single_user_api_key: str = secrets.token_hex(32)
+    single_user_api_key_generated: bool = "TILED_SINGLE_USER_API_KEY" not in os.environ
+    # Must be a string representation of a json list, e.g. ["one", "two"]
+    # The first key will be used for encryption. Each key will be tried in turn for decryption.
+    secret_keys: List[str] = [secrets.token_hex(32)]
```

Comment:
Another breaking change- this would currently be TILED_SECRET_KEYS instead of TILED_SERVER_SECRET_KEYS